modules/apps/blogs/blogs-web/src/main/java/com/liferay/blogs/web/internal/servlet/taglib/util/BlogsEntryActionDropdownItemsProvider.java

```diff
@@ -314,11 +316,17 @@ private String _getPermissionsURL(BlogsEntry blogsEntry) {
 	}
 
 	private String _getRedirectURL() {
-		return PortletURLBuilder.createRenderURL(
-			_renderResponse
-		).setMVCRenderCommandName(
-			"/blogs/view"
-		).buildString();
+		String redirect = ParamUtil.getString(_httpServletRequest, "redirect");
+
+		if (Validator.isBlank(redirect)) {
+			redirect = PortletURLBuilder.createRenderURL(
+				_renderResponse
+			).setMVCRenderCommandName(
+				"/blogs/view"
+			).buildString();
+		}
+
+		return redirect;
```

Comment:
Can you refactor this to use early returns?
```
String redirect = ParamUtil.getString(_httpServletRequest, "redirect");

if (Validator.isNotNull(redirect)) {
    return redirect;
}

return PortletURLBuilder.createRenderURL(...);
```

Reply:
In general, try to avoid code like:
```
int myMethod() {
   int someState = ...;

   // A ton of code that manipulates `someState`.

   return someState;
}
```
Although in this particular case it is not a problem (the method is tiny), this style makes it more difficult to understand what a function does. Essentially, if forces you to keep track of all possible state changes for a particular input, which can be difficult at times.
It is ok to use this style as long as the result is significantly simpler than other alternatives.

Reply:
Thanks @adolfopa, I updated the code.